third_party/parser/include/ruby_parser/driver.hh

```diff
@@ -154,13 +154,71 @@ public:
     }
 };
 
+class max_numparam_stack {
+    std::vector<int> stack;
+
+    friend class base_driver;
+
+public:
+    max_numparam_stack() = default;
+
+    void set_ordinary_params() {
+        set(-1);
+    }
+
+    bool has_ordinary_params() {
+        return top() < 0;
+    }
+
+    bool has_numparams() {
+        return top() > 0;
+    }
```

Comment:
What does it mean if `top()` == 0?

Reply:
I guess `0` is "either block that takes no params, or block that takes numparams but we haven't seen the first numparam yet"?

I'd be down to change these methods to `seen_*` to make it a little more clear what's happening, and/or add a comment about what it means if the top of the stack has `0`.

Reply:
> What does it mean if `top() == 0`?

It means we haven't parsed anything yet (the stack is empty).

Reply:
> I'd be down to change these methods to seen_* to make it a little more clear what's happening, and/or add a comment about what it means if the top of the stack has 0.

I reused the names from the original implementation (with minor tweaking because of the `?` question mark) but I'm more than happy to change them 👍